### PR TITLE
Add link for details for state mismatch exception

### PR DESCRIPTION
**Pull request by neha-bhargava**
#1872

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/MsalErrorMessage.cs [hunk 1]
```diff
@@ -97,6 +97,8 @@ public static string iOSBrokerKeySaveFailed(string keyChainResult)
             +
             "2. An error occurred during a silent web authentication that prevented the http authentication flow from completing in a short enough time frame";
 
+        public const string StateMismatchErrorMessage = "Returned state({0}) from authorize endpoint is not the same as the one sent({1}). See https://aka.ms/msal-statemismatcherror for more details.";
```
Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/MsalErrorMessage.cs:100]: Is it good from a security point of view to display the state that was sent?

**Review by jmprieur — COMMENTED** [review]
LGTM.
Thanks @neha-bhargava

**Review by bgavrilMS — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/MsalErrorMessage.cs (hunk 1 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/MsalErrorMessage.cs:100]: I don't think it would ,matter as it has already failed. @bgavrilMS Suggestions?

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/MsalErrorMessage.cs:100]: that's a good point, plus what would you do w/those values? and they are pretty long. i think just having the error message w/the link is enough.

Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/MsalErrorMessage.cs:100]: @jennyf19 Those were already there in the message. I just added the aka.ms link and moved the message to this class. I don't know what they were added for.

**Comment by henrik-me** [comment]
Looks great! Provided a comment in the issue about where the aka.ms link and the information should be placed

---
In reply to: [503660297](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2080#issuecomment-503660297) [](ancestors = 503660297)

**Review by henrik-me — APPROVED** [review]
:shipit:

**Comment by neha-bhargava** [comment]
> Looks great! Provided a comment in the issue about where the aka.ms link and the information should be placed
> 
> In reply to: [503660297](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2080#issuecomment-503660297) [](ancestors = 503660297)

Updated the Link.

**Pull request merged**